native/Avalonia.Native/src/OSX/main.mm

```diff
@@ -133,6 +138,26 @@ virtual HRESULT SetShowInDock(int show)  override
         }
     }
     
+    virtual HRESULT SetPreventAppNap(bool preventAppNap) override
+    {
+        START_COM_CALL;
+        
+        @autoreleasepool
+        {
+            if (preventAppNap && s_preventAppNapHandle == nullptr)
+            {
+                NSActivityOptions options = NSActivityUserInitiatedAllowingIdleSystemSleep;
+                s_preventAppNapHandle = [[NSProcessInfo processInfo] beginActivityWithOptions:options reason:@"User requested to prevent App Nap."];
```

Comment:
Something is wrong here.
When I assign to the "static" (in anonymous namespace here but that is equivalent) `s_preventAppNapHandle` I get a `DllNotFoundException` for `libAvaloniaNative` when the app starts. When `s_preventAppNapHandle` is a local variable this is not the case, and when I do not assign to `s_preventAppNapHandle` but still call `beginActivityWithOptions` then there it also loads.

Making `s_preventAppNapHandle` a static class variable has the same problem.

The thing is that `s_preventAppNapHandle` needs to be something static or in a singleton because as soon as this object is deallicated App Nap will turn on again.